### PR TITLE
[Backport 10.3] [ShaderGraph] [bugfix 1283902] Pull textures into Export Package even if not sampled by shader

### PR DESCRIPTION
**Pull request by cdxntchou**
### Purpose of this PR
Backport to 10.x of https://github.com/Unity-Technologies/Graphics/pull/2333
Includes additional backport of fixes in https://github.com/Unity-Technologies/Graphics/pull/2549

Fix for https://fogbugz.unity3d.com/f/cases/1283902/

All textures referenced by Texture asset nodes and texture input slots, whether actually used or not, will now be pulled into any Exported Package.

This includes all four texture type assets: Texture2D, Texture3D, Cubemap and Texture2DArray.

---
### Testing status
Describe what manual/automated tests were performed for this PR

ABV:   🟢 
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/10.x.x%252Fsg%252Ffix%252F1283902/.yamato%252F_abv.yml%2523all_project_ci_2020.2/4121735/job/pipeline

10.x.x/release ABV:  🟢 
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/10.x.x%252Frelease/.yamato%252F_abv.yml%2523all_project_ci_2020.2/4074558/job/pipeline

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
Could you test the following and take images showing they result in the texture being part of the exported package:
Texture2D, Texture2DArray, Texture3D, Cubemap, and Virtual Texture properties
Texture2D, Texture2DArray, Texture3D and Cubemap inline nodes
Inline texture inputs of Sample Texture 2D, Sample Texture3D, Sample Cubemap, Sample Texture 2D Array (not connected to the master stack)

**Review by ghost — APPROVED** [review]
I had some free time so I ran the manual tests for you.
image
image
image
Approved.

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed an issue where unknown type Nodes (i.e. HDRP-only nodes used without HDRP package) could be copied, resulting in an unloadable graph [1288475]
 - Fixed an issue where dropping HDRP-only properties from the blackboard field into the graph would soft-lock the graph [1288887]
 - Fixed Parallax Occlusion Mapping not working in sub graphs. [1221317](https://issuetracker.unity3d.com/product/unity/issues/guid/1221317/)
+- All textures in a ShaderGraph, even those not used, will now be pulled into an Exported Package [1283902]
```
Comment (marctem) [inline com.unity.shadergraph/CHANGELOG.md:36]: Changelog will need to move to new 10.3 section once 10.2 package is complete and we can merge this.

**Review by marctem — CHANGES_REQUESTED** [review]
All good except for changelog.

**Review by marctem — APPROVED** [review]

**Comment by ellioman** [comment]
Remember to include the warning fixes you made in #2549 in this PR

**Pull request merged**